### ISSUE TITLE
Call to an undefined method object::toArray() on Eloquent collection child within foreach loop

### ISSUE THREAD

**Issue by michaelsabatino** [nunomaduro/larastan#358]
- Larastan Version: ^0.4.3
- `--level` used: 2 and higher

### Description
Hi, Larastan issues this message:

Call to an undefined method object::toArray().

When foreach-ing through an Eloquent Collection and applying the toArray() function to the current child within the loop. The Laravel code executes without any issue, meaning the toArray() method works as expected.

### Laravel code where the issue was found

This defines the Eloquent relationship within the model:

```php

namespace App;

use Illuminate\Database\Eloquent\Model;
use App\Allocation;

class Form extends Model
{
    public function allocations()
    {
        return $this->belongsToMany(Allocation::class)->using('App\AllocationForm')->withPivot('optgroup');
    }
}
```

This causes the error message within the same model:

```php
foreach ($this->allocations as $allocation) :
    $myAllocation[] = $allocation->toArray();
endforeach;
```

However, the following does not cause the error message in Larastan and also works as desired.

```php
$i=0;
foreach ($this->allocations as $allocation) :
    $myAllocation[] = $this->allocations[$i]->toArray();
    $i++;
endforeach;
```
Please let me know if you need any further information.

**Comment by canvural** [comment]
Can you try adding `@property Collection|Allocation[] $allocations` to your `Form` model ? This can be a quick fix.

But I think we can write an extension for this. It would check if the model has a method with the same name as the property that is being accessed. Then we can specify the return type.

**Comment by michaelsabatino** [comment]
> Can you try adding `@property Collection|Allocation[] $allocations` to your `Form` model ? This can be a quick fix.
> 
> But I think we can write an extension for this. It would check if the model has a method with the same name as the property that is being accessed. Then we can specify the return type.

Thank you. I've added this inside the class declaration which seems to make the error go away.
```php
/**
    * Needed to stop an error message in Larastan
    *
    * @var \Illuminate\Support\Collection|Allocation[] $allocations
    */
    protected $allocations;
```

**Comment by canvural** [comment]
Ok. Thank you!

I'll try to figure out if we can write an extension for this. It seems like we can't get the model from the relation, easily.

**Comment by canvural** [comment]
This should be fixed with #361
